components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIGatewayManager.java

```diff
@@ -152,16 +160,17 @@ public void publishToGateway(WebApp api, APITemplateBuilder builder, String tena
 	public void removeFromGateway(WebApp api, String tenantDomain) throws Exception {
 		for (Environment environment : environments) {
 			RESTAPIAdminClient client = new RESTAPIAdminClient(api.getId(), environment);
-			
-			if (client.getApi(tenantDomain) != null) {
+
+			if (client.getVersionedWebApp(tenantDomain) != null) {
 				if (debugEnabled) {
 					log.debug("Removing WebApp " + api.getId().getApiName() + " From environment " +
-					          environment.getName());
+									  environment.getName());
 				}
-				String operation ="delete";
-				client.deleteApi(tenantDomain);
-				undeployCustomSequences(api, tenantDomain,environment);
-				setSecurevaultProperty(api,tenantDomain,environment,operation);
+				String operation = "delete";
+				client.deleteNonVersionedWebApp(tenantDomain);
+				client.deleteVersionedWebApp(tenantDomain);
+				undeployCustomSequences(api, tenantDomain, environment);
+				setSecurevaultProperty(api, tenantDomain, environment, operation);
```

Comment:
`setSecurevaultProperty` method was already there, merely used it here (see 5 lines above)
`api` variable was already there, merely used it here (see 5 lines above)